library/alloc/src/boxed.rs

```diff
@@ -1616,7 +1616,6 @@ impl<T, const N: usize> From<[T; N]> for Box<[T]> {
     /// println!("{boxed:?}");
     /// ```
     fn from(array: [T; N]) -> Box<[T]> {
-        #[rustc_box]
         Box::new(array)
```

Comment:
Same as `Box::pin`, in an unoptimized build, `T` already needs to be on the stack, so this change shouldn't inhibit any in-place heap initialization.